dot/network/state.go

```diff
@@ -42,7 +42,7 @@ type Syncer interface {
 	CreateBlockResponse(*BlockRequestMessage) (*BlockResponseMessage, error)
 }
 
-//go:generate mockery --name TransactionHandler --structname MockTransactionHandler --case underscore --inpackage
+//go:generate mockgen -destination=mock_transaction_handler.go -package $GOPACKAGE . TransactionHandler
```

Comment:
change to 

```suggestion
//go:generate mockgen -destination=mock_transaction_handler_test.go -package $GOPACKAGE . TransactionHandler
```

Unless it's needed by other packages. Actually ideally I think generating package local mocks in `_test.go` files is best.

If you need i.e. to mock `io.Writer` for the tests in `some/package`, then have 

```go
//go:generate mockgen -destination=mock_writer_test.go -package $GOPACKAGE io Writer
```

Where you need it (in your test file for example).

That ensures the go API is not polluted by mocks and also makes them local where they're needed. It does introduce some generated code duplication though but I think it's worth it.

Reply:
These mocks need to be used by other packages, although agreed with you I think I will leave as it is to avoid too many changes in the codebase right now, but we can create an issue under Tests epic to handle the mocks

Reply:
Totally, thanks!